pkg/commatrix-creator/commatrix.go

```diff
@@ -41,6 +41,9 @@ func New(exporter *endpointslices.EndpointSlicesExporter, customEntriesPath stri
 // port number, namespace, service name, pod, container, node role, and flow optionality for OpenShift.
 func (cm *CommunicationMatrixCreator) CreateEndpointMatrix() (*types.ComMatrix, error) {
 	log := utils.Logger() // Get the logger instance from utils
+	if log == nil {
```

Comment:
Why do you check for nil? If a object using a logger it should init and hold a logger object